swap_meet/vendor.py

```diff
@@ -1,2 +1,116 @@
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+        
+    def add(self,item):
+        """This function takes an item and adds it to the inventory.
+        It then returns the item"""
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        """This function takes an item and removes the matching
+        item from the inventory. After removing the item, this 
+        function returns the removed item. If the item does not 
+        exist in the inventory, the function returns False."""
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+    def get_by_category(self, category):
+        """This function takes in a category and returns a list of
+        items in the inventory that matches the category passed."""
+        items = []
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+        return items
+    
+    def swap_items(self, friend, my_item, their_item):
+        """This function swaps an item in the Vendor's inventory
+        with another item in the friend's inventory. If the swap is 
+        successful, the function returns True. Otherwise, returns
+        False."""
+        if my_item in self.inventory and their_item in friend.inventory:
+            friend.inventory.append(my_item)
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            friend.inventory.remove(their_item)
```

Comment:
this works! but one thing we could do is utilize the instance methods we made above, like so:

```suggestion
            friend.add(my_item)
            self.remove(my_item)
            self.add(their_item)
            friend.remove(their_item)
```

we can go one step further and combine the `remove` call inside the `add`: `self.add(friend.remove(their_item))`. why do you think we can do that?